### ISSUE TITLE
URL parsing throws false negatives

### ISSUE THREAD

**Issue by kaelig** [t32k/stylestats#91]
I used the Regex found in Stylestats in a project of mine (to detect if a JavaScript file was passed as a URL), but it did not accept my url `https://cdn.polyfill.io/v1/polyfill.min.js?ua=firefox/30`.

The regex can be found in stylestats.js: https://github.com/t32k/stylestats/blob/master/lib/stylestats.js#L29
```js
var URL = /^(?:(?:ht|f)tp(?:s?)\:\/\/|~\/|\/)?(?:\w+:\w+@)?((?:(?:[-\w\d{1-3}]+\.)+(?:com|org|cat|coop|int|pro|tel|xxx|net|gov|mil|biz|info|mobi|name|aero|jobs|edu|co\.uk|ac\.uk|it|fr|tv|museum|asia|local|travel|[a-z]{2})?)|((\b25[0-5]\b|\b[2][0-4][0-9]\b|\b[0-1]?[0-9]?[0-9]\b)(\.(\b25[0-5]\b|\b[2][0-4][0-9]\b|\b[0-1]?[0-9]?[0-9]\b)){3}))(?::[\d]{1,5})?(?:(?:(?:\/(?:[-\w~!$+|.,=]|%[a-f\d]{2})+)+|\/)+|\?|#)?(?:(?:\?(?:[-\w~!$+|.,*:]|%[a-f\d{2}])+=?(?:[-\w~!$+|.,*:=]|%[a-f\d]{2})*)(?:&(?:[-\w~!$+|.,*:]|%[a-f\d{2}])+=?(?:[-\w~!$+|.,*:=]|%[a-f\d]{2})*)*)*(?:#(?:[-\w~!$ |\/.,*:;=]|%[a-f\d]{2})*)?$/;
```

I can recommend the use of the [valid-url](https://www.npmjs.org/package/valid-url) package which worked well for me.

**Comment by kamranayub** [comment]
Why is there even parsing? Issue a HEAD, request OK, sounds good; fallback to GET if HEAD is blocked. I tried pasting in a link and it's failing. Not a good first impression :(
